[Dart 3.8] Introduce documentation for doc imports (#6538)
dart-lang/site-www: src/content/tools/doc-comments/references.md

The doc comment for a type alias that aliases a class, enum, extension type, or
mixin can't reference any of the aliased type's members as if they were in
scope.

## Doc imports

Dart provides a `@docImport` documentation tag, which enables external elements
to be referenced in doc comments without actually importing them. This tag can be
specified in a doc comment above a `library` directive. For example:

```dart
/// @docImport 'dart:async';
library;

/// We can now reference elements like [Future] and [Future.value] from
/// dart:async, even if the library is not imported with an actual import.
class Foo {}
```

> Doc imports support all of the same URI styles as regular Dart imports (for example,

Author: Thanks! Done

The doc comment for a type alias that aliases a class, enum, extension type, or
mixin can't reference any of the aliased type's members as if they were in
scope.

> ## Documentation imports

Reviewer: ```suggestion
## Doc imports
```

> Dart provides a "documentation imports" feature which enables external elements

Reviewer: I would rewrite as:

```
Dart provides a `@docImport` documentation tag, which enables external elements
to be referenced in doc comments without actually importing them. This tag can be
specified in a doc comment above a `library` directive. For example:
```

Author: I love it. Done.

to be referenced in doc comments without actually importing them via a Dart
import directive. Doc imports are specified in a doc comment on a `library`
directive. For example:

```dart
/// @docImport 'dart:async';
library;

/// We can now reference elements like [Future] and [Future.value] from
/// dart:async, even if the library is not imported with an actual import.
class Foo {}
```

> Doc imports support all of the same URI styles as regular Dart imports (e.g.

Reviewer: ```suggestion
Doc imports support all of the same URI styles as regular Dart imports (for example,
```

'dart:', 'package:', and relative paths). They can't contain the `deferred`

> keyword, or configurations. Doc imports currently can't contain `as`, `show`,

Reviewer: ```suggestion
keyword or configurations. Doc imports currently can't contain `as`, `show`,
```

or `hide`.

> :::version-note

Reviewer: I would delete this:

```
:::version-note
Doc imports were introduced in Dart 3.8.
:::
```

Instead, let's not submit this PR until the Dart 3.8 release.

Reviewer 2: Up to you, but personally I prefer to keep version notes for a while. Developers often stay on an older version of Dart for a while and without a note, they might get confused when a feature doesn't work.

Plus a note allows us to land relevant docs early and benefit developers using `beta` releases (which we want them to do) :D

Author: I agree, I think there are many users that stay on one stable version behind or even two or three, if they work on bigger apps.
